python/pyspark/pandas/sql_formatter.py

```diff
@@ -265,7 +266,14 @@ def _convert_value(self, val: Any, name: str) -> Optional[str]:
             val._to_spark().createOrReplaceTempView(df_name)
             return df_name
         elif isinstance(val, str):
-            return lit(val)._jc.expr().sql()  # for escaped characters.
+            if is_remote():
+                special_characters = ["\\", "'", '"', "\0", "\b", "\n", "\r", "\t"]
```

Comment:
let's exactly match `"'" + v.toString.replace("\\", "\\\\").replace("'", "\\'") + "'"` in the JVM side

Reply:
Sure. let me update the comments. Thanks!